torture.c

```diff
@@ -8,18 +8,92 @@
 #include <stdlib.h>
 #include <time.h>
 #include <unistd.h>
+#include <string.h>
 #include <math.h>
 
+const char help[] =
+    "Usage:\n"
+    "  torture [-2] [-c] [-g] [-r rate]\n"
+    "  torture -h\n"
+    "\n"
+    "  -h       print this help message and exit\n"
+    "  -2       output two waves\n"
+    "  -c       randomly chunk the output\n"
+    "  -g       occasionally output garbage\n"
+    "  -r rate  sample rate in samples/s (default: 100)\n"
+    "  -s seed  set random seed\n";
+
+const char optstring[] = "h2cgr:s:";
+
 int main(int argc, char *argv[]) {
-    (void) argv;
-    unsigned int n;
-
-    for(n=0;;n+=5) {
-        printf("%.1f\n", (sin(n*M_PI/180)*5)+5);
-        if(argc==2)
-            printf("%.1f\n", (cos(n*M_PI/180)*5)+5);
-        fflush(stdout);
-        usleep(10000);
+    char buffer[1024];
+    size_t buffer_pos = 0;
+    int opt;
+    int two_waves = 0;
+    int chunked = 0;
+    int add_garbage = 0;
```

Comment:
I noticed only now we have three more bools in disguise here. Can we turn these to stdbool also?